dotrix_pbr/src/shaders/skeletal.wgsl

```diff
@@ -89,30 +89,79 @@ fn vs_main(
 
 // STAGE: FRAGMENT -------------------------------------------------------------------------------
 struct Material {
-    albedo: vec4<f32>;
-    has_texture: u32;
+  color: vec4<f32>;
+  has_texture: u32;
+};
+struct Materials {
+    albedo: Material;
+    roughness: Material;
+    metallic: Material;
+    ao: Material;
```

Comment:
What is the reason behind this layout? What do you think about to optimize it like this
```
struct Material {
    color: vec4<f32>;
    has_texture: vec4<u32>; // x - albedo, y - roughness, z - metallic, w - ao or even pack them in bits
    roughness: vec4<f32>;
    metallic: vec4<f32>;
    ao: vec4<f32>;
```

Reply:
We could pack it like this too

```glsl
struct Material {
    color: vec4<f32>;
    has_texture: vec4<u32>; // x - albedo, y - roughness, z - metallic, w - ao or even pack them in bits
    properties: vec4<f32>; // x - albedo, y - roughness, z - metallic, w - ao
};

Reply:
I was afraid to suggest it lol :) but that would be the best

Reply:
I'll try bitwise but my bit math might need some work

Reply:
Ok compact materials is in